### PR TITLE
Update components to use sprkLink directive

### LINKED ISSUE #2893: Port over Spark components to use new `sprkLink` directive
**User Story:**

**Notes:**

**AC:**
- port all components over to use new sprkLink
- Components:
pagination 
promo
toggle
- run test & make adjustments where needed in our test
- verify there are no changes to interfaces or masthead dropdown interfaces.

### PR DESCRIPTION
**Pull request by afebbraro**
## What does this PR do?
A clear and concise description of what this PR does.

### Associated Issue
Fixes #2893 

## Please check off completed items as you work.
If a checklist item or section does not apply to your PR
then please remove it.

### Documentation
 - [x] Update Spark Docs

### Code
 - [x] Build Component in Angular
 - [x] Unit Testing in Angular with `npm run test` in `angular/` (100% coverage, 100% passing)

### Accessibility
- [x] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [ ] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [ ] Mozilla Firefox (Mobile)
  - [ ] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [ ] Microsoft Edge
  - [x] Apple Safari
  - [ ] Apple Safari (Mobile)

### CODE REVIEW

File: angular/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "0.0.0",
   "scripts": {
     "build": "npm run set-version && ng build spark-angular",
-    "test": "jest --config ./projects/spark-angular/jest.config.js",
+    "test": "jest --config ./projects/spark-angular/jest.config.js --verbose=false",
```
Comment (afebbraro) [inline angular/package.json]: Adding in the verbose flag here makes it so console.logs show up. Otherwise, they don't.

Comment (RCopeland) [reply angular/package.json]: dont need the =false

File: angular/projects/spark-angular/src/lib/components/sprk-accordion-item/sprk-accordion-item.component.ts [hunk 2]
```diff
@@ -29,7 +31,7 @@ import { toggleAnimations } from '../sprk-toggle/sprk-toggle-animations';
           }}"
           [iconType]="currentIconType"
         ></sprk-icon>
-      </sprk-link>
+        </a>
```
Comment (guern1kn) [inline angular/projects/spark-angular/src/lib/components/sprk-accordion-item/sprk-accordion-item.component.ts]: Extra space

File: angular/projects/spark-angular/src/lib/components/sprk-card/sprk-card.component.ts [hunk 3]
```diff
@@ -8,19 +8,20 @@ import { Component, Input } from '@angular/core';
       *ngIf="cardType === 'teaser'"
       [attr.data-id]="idString"
     >
-      <sprk-link
-        linkType="unstyled"
+      <a
+        sprkLink
+        variant="unstyled"
         *ngIf="media === 'img'"
         [href]="imgHref"
```
Comment (guern1kn) [inline angular/projects/spark-angular/src/lib/components/sprk-card/sprk-card.component.ts]: [attr.href]

File: angular/projects/spark-angular/src/lib/components/sprk-dropdown/sprk-dropdown.component.ts [hunk 4]
```diff
@@ -33,7 +35,7 @@ import { ISprkDropdownChoice } from './sprk-dropdown.interfaces';
               additionalIconClasses
             }}"
           ></sprk-icon>
-        </sprk-link>
+          </a>
```
Comment (guern1kn) [inline angular/projects/spark-angular/src/lib/components/sprk-dropdown/sprk-dropdown.component.ts]: Extra space

**Pull request merged**